hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1257,6 +1257,35 @@ Y(incwm, incw, loadw, storew, m)
 
 #undef Y
 
+void lower(Vunit& unit, cvttsd2siq& i, Vlabel b, size_t idx) {
+  lower_impl(unit, b, idx, [&] (Vout& v) {
+  auto fpsr = v.makeReg();
+  auto res = v.makeReg();
+  auto const sf = v.makeReg();
+  auto err = v.makeReg();
+  auto tmp1 = v.makeReg();
+  auto tmp2 = v.makeReg();
+
+  // Clear FPSR IOC flag
+  v << mrs{FPSR, tmp1};
+  v << andqi{~0x01, tmp1, tmp2, sf};
+  v << msr{tmp2, FPSR};
+
+  // Load error value
+  v << ldimmq{0x8000000000000000, err};
+
+  // Do ARM64's double to signed int64 conversion.
+  v << fcvtzs{i.s, res};
+
+  // Check if there was a conversion error
+  v << mrs{FPSR, fpsr};
+  v << testqi{1, fpsr, sf};
+
+  // Move converted value or error
+  v << cmovq{CC_NZ, sf, res, err, i.d};
+  });
```

Comment:
While this lowering is certianly fine for getting the port working, the HHIR->VASM lowering code for double->int conversion is probably too dependent on the behavior of X64's cvttsd2siq for the edge cases.

IMHO it would be acceptable to do an arch() check before emitting cvttsd2siq, and just generate different code to check/handle the error conditions. (but cc @mxw)

a nicer solution would be to define a higher level vasm instruction then lower it customly for x64/ppc/arm, however I don't have any good suggestions for how to define that instruction. it could encapsulate *all* the error handling, but that feels too high-level and PHP dependent. and it seems hard to abstractly define the way it would signal errors.